syntax = "proto3";

package rpc;

service Cache {
    rpc Store(StoreReq) returns (StoreResp) {}
    rpc Get(GetReq) returns (GetResp) {}
}

message StoreReq {
    string key = 1;
    bytes val = 2;
    string account_token = 3;
}

message StoreResp {
}

message GetReq {
    string key = 1;
}

message GetResp {
    bytes val = 1;
}

service StreamVideo {
    rpc GetVideo(Empty) returns (stream Image) {}
}

message Empty {}

message Image {
    bytes image = 1;
}

service Accounts {
    rpc GetByToken(GetByTokenReq) returns (GetByTokenResp) {}
}

message GetByTokenReq {
    string token = 1;
}

message GetByTokenResp {
    Account account = 1;
}

message Account {
    int64 max_StreamVideo_keys = 1;
}